frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js

```diff
@@ -0,0 +1,103 @@
+
+    // Stryker disable all
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+    } = useForm(
+        { defaultValues: initialPersonalSchedule || {}, }
+    );
+    // Stryker enable all
+
+    const navigate = useNavigate();
+
+    const yyyyq_regex = /((19)|(20))\d{2}[1-4]/i;
+
+    return (
+
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialPersonalSchedule && (
+                <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="id">Id</Form.Label>
+                    <Form.Control
+                        data-testid="PersonalScheduleForm-id"
+                        id="id"
+                        type="text"
+                        {...register("id")}
+                        value={initialPersonalSchedule.id}
+                        disabled
+                    />
+                </Form.Group>
+            )}
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="name">Name</Form.Label>
+                <Form.Control
+                    data-testid="PersonalScheduleForm-name"
+                    id="name"
+                    type="text"
+                    isInvalid={Boolean(errors.name)}
+                    {...register("name", { required: "Name is required." })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.name?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="description">Description</Form.Label>
+                <Form.Control
+                    data-testid="PersonalScheduleForm-description"
+                    id="description"
+                    type="text"
+                    isInvalid={Boolean(errors.description)}
+                    {...register("description", { required: "Description is required." })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.description?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="quarterYYYYQ">Quarter YYYYQ</Form.Label>
+                <Form.Control
+                    data-testid="PersonalScheduleForm-quarterYYYYQ"
+                    id="quarterYYYYQ"
+                    type="text"
+                    isInvalid={Boolean(errors.quarterYYYYQ)}
+                    {...register("quarterYYYYQ", { required: true, pattern: yyyyq_regex })}
+                />
+                <Form.Control.Feedback type="invalid">
+                {errors.quarterYYYYQ && 'QuarterYYYYQ is required. '}
+                {errors.quarterYYYYQ?.type === 'pattern' && 'QuarterYYYYQ must be in the format YYYYQ, e.g. 20224 for Fall 2022'}
+                </Form.Control.Feedback>
+            </Form.Group>
```

Comment:
Rather than have the user enter the quarter in YYYYQ format, which is unfamilar to most users, it is probably better to use the preexisting `SingleQuarterDropdown.js` component.

https://github.com/ucsb-cs156-w22/team04-w22-6pm-courses/blob/main/frontend/src/main/components/Quarters/SingleQuarterDropdown.js

That way, the value is always in the correct format, but what the users sees is what they are familiar with: F20, W19, etc.

Reply:
How do I use that component in my code? I don't really know what to do with it or how to implement it into my PersonalScheduleForm.js